spec/functions/lookup_spec.rb

```diff
@@ -94,4 +94,17 @@
       expect(result.unwrap).to eq('foo' => 'bar')
     end
   end
+
+  it 'logs on on Vault with a namespace, requests a secret using a token, and returns the data wrapped in the Sensitive type' do
+    vault_server = MockVault.new
+    vault_server.mount('/v1/auth/cert/login', AuthSuccess)
+    vault_server.mount('/v1/kv/test', SecretLookupSuccess)
+    vault_server.start_vault do |port|
+      stub_const('ENV', ENV.to_hash.merge('VAULT_ADDR' => "http://127.0.0.1:#{port}"))
+      stub_const('ENV', ENV.to_hash.merge('VAULT_NAMESPACE' => 'foo'))
+      result = function.execute('kv/test')
+      expect(result).to be_a(Puppet::Pops::Types::PSensitiveType::Sensitive)
+      expect(result.unwrap).to eq('foo' => 'bar')
+    end
+  end
```

Comment:
Here we should at least check that `X-Vault-Namespace` header exists when we use either the ENV or the param.

I'm not sure how this can be done with the new test suit :(

Reply:
You'll want to create a class that is mounted on the `vault_server` that introspects the request to ensure that it has the data expected. For example, [AuthSuccessWithRole](https://github.com/voxpupuli/puppet-vault_lookup/blob/master/spec/mock_vault_helper.rb#L78) does this by ensuring `request.body` matches the expected data for the body.

I expect that the class to mount mount would look something like this:
```ruby
  class ExpectingSpecialHeader < WEBrick::HTTPServlet::AbstractServlet
    def do_POST(request, response)
      if request.header['VAULT-NAMESPACE'] == ['foo']
        response.status = 200
        response.content_type = 'application/json'
      else
        response.status = 403
      end
    end
  end
```

Reply:
Done. Thanks @tvpartytonight  for the suggestion !